To recap, the message in 2.13.12 is imprecise and has a coarse position.
```
test/files/neg/t12813c.scala:6: error: a is renamed twice
import O.{a => abcdefghij, a => abcdefghij}
       ^
1 error
```

The updated message makes it clear, hopefully, that what's significant is the right of the arrows. (`import O.x` means `import O.{x => x}`) What matters is that two selectors introduce an ambiguity.

I wound up quoting the entire check file here. The possible "improvement" would be to notice line 8 that left of the arrows are also identical.
```
t12813.scala:5: error: a is imported twice
import O.{a, a}   // error
          ^
t12813.scala:8: error: b is an ambiguous name on import
import O.{a => b, a => b}   // error
               ^
t12813.scala:10: error: b is an ambiguous name on import
import O.{a => b, toString => b}    // error
               ^
t12813.scala:11: error: toString is an ambiguous name on import
import O.{a => toString, toString}  // error
               ^
t12813.scala:12: error: toString is an ambiguous name on import
import O.{toString, a => toString}  // error
          ^
5 errors
```

--- a/src/compiler/scala/tools/nsc/ast/parser/Parsers.scala
+++ b/src/compiler/scala/tools/nsc/ast/parser/Parsers.scala
@@ -2825,9 +2825,11 @@ self =>
             if (!h.isMask)
               t.find(_.rename == h.rename).foreach { duplicate =>
                 val msg =
-                  if (h.isRename || duplicate.isRename) s"${h.rename} is an ambiguous name on import"
+                  if (h.isRename || duplicate.isRename)
+                    if (h.name == duplicate.name) s"${h.name} is renamed twice to ${h.rename}"
+                    else s"${h.rename} is an ambiguous name on import"
                   else s"${h.rename} is imported twice"
-                syntaxError(h.renamePos, msg)
+                syntaxError(duplicate.renamePos, msg)
               }
             h :: checkSelectors(t)
           }

--- a/src/compiler/scala/tools/nsc/ast/parser/Parsers.scala
+++ b/src/compiler/scala/tools/nsc/ast/parser/Parsers.scala
@@ -2827,7 +2827,7 @@ self =>
                 val msg =
                   if (h.isRename || duplicate.isRename) s"${h.rename} is an ambiguous name on import"
                   else s"${h.rename} is imported twice"
-                syntaxError(h.namePos, msg)
+                syntaxError(h.renamePos, msg)
               }
             h :: checkSelectors(t)
           }
@@ -2852,7 +2852,8 @@ self =>
         if (in.token == ARROW || (currentRun.isScala3 && isRawIdent && in.name == nme.as)) {
           in.nextToken()
           if (name == nme.WILDCARD && !bbq) syntaxError(in.offset, "Wildcard import cannot be renamed")
-          (wildcardOrIdent(), in.offset)
+          val pos = in.offset
+          (wildcardOrIdent(), pos)
         }
         else if (name == nme.WILDCARD && !bbq) (null, -1)
         else (name, start)

--- a/test/files/neg/t12813.check
+++ b/test/files/neg/t12813.check
@@ -1,16 +1,16 @@
 t12813.scala:5: error: a is imported twice
 import O.{a, a}   // error
-          ^
-t12813.scala:8: error: b is an ambiguous name on import
+             ^
+t12813.scala:8: error: a is renamed twice to b
 import O.{a => b, a => b}   // error
-               ^
+                       ^
 t12813.scala:10: error: b is an ambiguous name on import
 import O.{a => b, toString => b}    // error
-               ^
+                              ^
 t12813.scala:11: error: toString is an ambiguous name on import
 import O.{a => toString, toString}  // error
-               ^
+                         ^
 t12813.scala:12: error: toString is an ambiguous name on import
 import O.{toString, a => toString}  // error
-          ^
+                         ^
 5 errors

--- a/test/files/neg/t12813.check
+++ b/test/files/neg/t12813.check
@@ -3,13 +3,13 @@ import O.{a, a}   // error
           ^
 t12813.scala:8: error: b is an ambiguous name on import
 import O.{a => b, a => b}   // error
-          ^
+               ^
 t12813.scala:10: error: b is an ambiguous name on import
 import O.{a => b, toString => b}    // error
-          ^
+               ^
 t12813.scala:11: error: toString is an ambiguous name on import
 import O.{a => toString, toString}  // error
-          ^
+               ^
 t12813.scala:12: error: toString is an ambiguous name on import
 import O.{toString, a => toString}  // error
           ^

--- a/test/files/neg/t12813b.check
+++ b/test/files/neg/t12813b.check
@@ -3,13 +3,13 @@ import O.{a, a}   // error
           ^
 t12813b.scala:8: error: b is an ambiguous name on import
 import O.{a => b, a => b}   // error
-          ^
+               ^
 t12813b.scala:10: error: b is an ambiguous name on import
 import O.{a => b, toString => b}    // error
-          ^
+               ^
 t12813b.scala:11: error: toString is an ambiguous name on import
 import O.{a => toString, toString}  // error
-          ^
+               ^
 t12813b.scala:12: error: toString is an ambiguous name on import
 import O.{toString, a => toString}  // error
           ^

--- a/test/files/neg/t12813b.check
+++ b/test/files/neg/t12813b.check
@@ -1,18 +1,18 @@
 t12813b.scala:5: error: a is imported twice
 import O.{a, a}   // error
-          ^
-t12813b.scala:8: error: b is an ambiguous name on import
+             ^
+t12813b.scala:8: error: a is renamed twice to b
 import O.{a => b, a => b}   // error
-               ^
+                       ^
 t12813b.scala:10: error: b is an ambiguous name on import
 import O.{a => b, toString => b}    // error
-               ^
+                              ^
 t12813b.scala:11: error: toString is an ambiguous name on import
 import O.{a => toString, toString}  // error
-               ^
+                         ^
 t12813b.scala:12: error: toString is an ambiguous name on import
 import O.{toString, a => toString}  // error
-          ^
+                         ^
 t12813b.scala:14: error: wildcard import must be in last position
 import O.{given, a, _}    // error 3
           ^

--- a/test/files/neg/t12813c.check
+++ b/test/files/neg/t12813c.check
@@ -1,4 +1,4 @@
-t12813c.scala:6: error: abcdefghij is an ambiguous name on import
+t12813c.scala:6: error: a is renamed twice to abcdefghij
 import O.{a => abcdefghij, a => abcdefghij}
-               ^
+                                ^
 1 error

--- a/test/files/neg/t12813c.check
+++ b/test/files/neg/t12813c.check
@@ -1,0 +1,4 @@
+t12813c.scala:6: error: abcdefghij is an ambiguous name on import
+import O.{a => abcdefghij, a => abcdefghij}
+               ^
+1 error

--- a/test/files/neg/t12813c.scala
+++ b/test/files/neg/t12813c.scala
@@ -1,0 +1,6 @@
+
+//> abusing options -Vprint:parser -Vpos -Yprint-trees:format
+
+object O { val a = 42 }
+
+import O.{a => abcdefghij, a => abcdefghij}

